### PR TITLE
Reorganize the API documentation.

### PR DESCRIPTION
**Pull request by jsiirola**
Split the API documentation to differentiate between core functionality, beta functionality, and third-party documentation.

## Summary/Motivation:
We are starting to get user questions that reference the online documentation (a good thing!); however, they were referencing the Kernel documentation and implying that it would work for general Pyomo models (https://stackoverflow.com/questions/48538945/access-all-variables-occurring-in-a-pyomo-constraint).  This change highlights that the Kernel is still "beta" -- mostly because that API should not (yet) be expected to work with models that general users would be creating.

## Changes proposed in this PR:
- Reorganize the TOC for the Library Reference to split out beta and third-party documentation.

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the BSD license.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

**Comment by qtothec** [comment]
It'd be nice if read_the_docs also built relevant PRs.

**Comment by jsiirola** [comment]
@qtothec: agreed.  Unfortunately, we haven't been able to figure out how to make that happen automatically.  There appear to be issues tracking this feature request for RTD (rtfd/readthedocs.org#1340 and rtfd/readthedocs.org#2465)

**Comment by blnicho** [comment]
If you view the full .rst file on github it gives you a slightly better preview of what it will look like than just looking at the raw text (minus the special linking done by Sphinx)

File: doc/OnlineDocs/library_reference/contrib.rst [hunk 1]
```diff
@@ -1,5 +1,21 @@
 Third-Party Contributions
 =========================
 
-* `pyomo.contrib.simplemodel <http://pyomocontrib-simplemodel.readthedocs.io/en/latest/>`_
+Pyomo includes a variety of additional features and functionality
+provided by thiry parties through the ``pyomo.contrib`` package.  This
```
Comment (qtothec) [inline doc/OnlineDocs/library_reference/contrib.rst]: "third" parties

**Comment by jsiirola** [comment]
Spoke with @carldlaird, @blnicho, and @whart222: We agreed to merge this to clarify the Library docs, but hold off on any additional changes until after the 5.4 release and #272 is merged (which has a more extensive rework of the documentation hierarchy - in particular to distinguish between User API docs and Developer API docs).

**Pull request merged**